### PR TITLE
More Postprocesses ported to webgpu

### PR DESCRIPTION
**Pull request by deltakosh**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15447/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15447/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15447/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15447/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15447/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15447/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15447/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15447/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15447/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15447/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGL 1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15447/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by deltakosh** [comment]
Moving to draft as I will reconsider how pp transfer their imports to avoid the current braking change

File: packages/dev/core/src/Layers/highlightLayer.ts [hunk 1]
```diff
@@ -79,15 +79,15 @@ class GlowBlurPostProcess extends PostProcess {
...
 
-    protected override async _initShaderSourceAsync(useWebGPU: boolean) {
+    protected override _gatherImports(useWebGPU: boolean, list: Promise<any>[]) {
         if (useWebGPU) {
             this._webGPUReady = true;
-            await import("../ShadersWGSL/glowBlurPostProcess.fragment");
+            list.push(import("../ShadersWGSL/glowBlurPostProcess.fragment"));
         } else {
-            await import("../Shaders/glowBlurPostProcess.fragment");
+            list.push(import("../Shaders/glowBlurPostProcess.fragment"));
```
Comment (deltakosh) [inline packages/dev/core/src/Layers/highlightLayer.ts:87]: @RaananW I hope this is ok for our build?

**Review by sebavan — APPROVED** [review]

File: packages/dev/core/src/Layers/highlightLayer.ts (hunk 1 above)
Comment (RaananW) [reply packages/dev/core/src/Layers/highlightLayer.ts:87]: yes, as long as there is `import("...")` in one line it is fine

**Pull request merged**